src/context.rs

```diff
@@ -579,6 +584,30 @@ impl PySessionContext {
             Err(err) => Ok(format!("Error: {:?}", err.to_string())),
         }
     }
+
+    /// Execute a partition of an execution plan and return a stream of record batches
+    pub fn execute(
+        &self,
+        plan: PyExecutionPlan,
+        part: usize,
+        py: Python,
+    ) -> PyResult<PyRecordBatchStream> {
+        let ctx = Arc::new(TaskContext::new(
+            "task_id".to_string(),
```

Comment:
Created an issue for more general configuration setup https://github.com/apache/arrow-datafusion-python/issues/187